config/config.go

```diff
@@ -104,8 +102,12 @@ func (*nullCloser) Close() error { return nil }
 // before shutdown.
 func (c Configuration) New(
 	serviceName string,
-	statsReporter jaeger.StatsReporter,
+	options ...ClientOption,
 ) (opentracing.Tracer, io.Closer, error) {
+	c.clientMetrics = jaeger.NewNullMetrics()
```

Comment:
usually you apply functional options first and then check if a value is nil and default it